clients/pkg/promtail/positions/positions.go

```diff
@@ -176,14 +179,20 @@ func (p *positions) save() {
 	}
 }
 
+// CursorKey returns a key that can be saved as a cursor that is never deleted.
+func CursorKey(key string) string {
+	return fmt.Sprintf("%s%s", cursorKeyPrefix, key)
+}
+
 func (p *positions) cleanup() {
 	p.mtx.Lock()
 	defer p.mtx.Unlock()
 	toRemove := []string{}
 	for k := range p.positions {
-		// If the position file is prefixed with journal, it's a
-		// JournalTarget cursor and not a file on disk.
-		if strings.HasPrefix(k, "journal-") {
+		// If the position file is prefixed with cursor, it's a
+		// cursor and not a file on disk.
+		// We still have to support journal files, so we keep the previous check to avoid breaking change.
+		if strings.HasPrefix(k, cursorKeyPrefix) || strings.HasPrefix(k, "journal-") {
```

Comment:
Should this be
```suggestion
		if strings.HasPrefix(k, cursorKeyPrefix) || strings.HasPrefix(k, journalKeyPrefix) {
```
for consistency?